README.md

```diff
@@ -1,3 +1,88 @@
 # ioc-hierarchy-service
 In-memory queryable hierarchy service
 
+This project generates a debian package with all components necessary to
+create an in-memory cache of a hierarchy. The hierarchy is modelled as
+a graph and can be queried in a number of ways. The cache itself is run
+as a daemon process which can be backed up by using another daemon
+process called backup service. All interprocess communication is handled
+with ZeroMQ.
+
+Querying the process must be done using the bundled python module. This
+can be imported and a function named `query` can be run, e.g. to check
+the health of the service:
+
+```
+>>> import tag_hierarchy_client
+>>> tag_hierarchy_client.query([{'command': 'healthcheck'}], "server")
+```
+
+# Architectural overview
+The following diagram describes which components are bundled with the
+debian package and how they are intended to be used:
+
+![Architecture diagram][docs/hierarchy_daemon.png]
+
```

Comment:
The diagram is brilliant. However, a couple of small clarifications could be made: 1) What does the arrows signify? Messages/Commands or Dependencies? Also the "tag hierarchy server" and "tag hierarchy statemanager" have identical descriptions. - should they not be different?

Reply:
Agree, will add some pointers. The processes are identical, just running in different modes, but that could probably be stated more clearly.

Reply:
Ah, just checked the diagram it is indeed wrong. Will fix.